client/network-gossip/src/bridge.rs

```diff
@@ -222,6 +164,59 @@ impl<B: BlockT> GossipEngine<B> {
 	}
 }
 
+impl<B: BlockT> Future for GossipEngine<B> {
+	type Output = ();
+
+	fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+		self.inner.lock().poll_unpin(cx)
+	}
+}
+
+impl<B: BlockT> Future for GossipEngineInner<B> {
+	type Output = ();
+
+	fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+		let this = &mut *self;
+
+		while let Poll::Ready(Some(Ok(event))) = this.network_event_stream.poll_next_unpin(cx) {
```

Comment:
This is copy and pasted from the previously spawned logic. In the long run I think we should at least log errors on this channel.